src/main/java/io/lettuce/core/ReadFrom.java

```diff
@@ -193,6 +195,22 @@ public static ReadFrom valueOf(String name) {
             throw new IllegalArgumentException("Name must not be empty");
         }
 
+        int index = name.indexOf(':');
+
+        if (index != -1) {
+            String type = name.substring(0, index);
+            String value = name.substring(index + 1);
+            if (LettuceStrings.isEmpty(value)) {
```

Comment:
Just for the sake of posterity could we also check for index == 0 to make sure we handle something like :192.0.2.0/24?

Reply:
added test

Reply:
:) funny, I was thinking we should add another check if `type` is empty, but when I saw your test it makes much more sense to leave it as you did (because we cant assume that the `:` was put there on purpose).

Thanks, that looks perfect.